typeinfo.go

```diff
@@ -267,6 +267,23 @@ func encodeBytesValue(e *Encoder, v reflect.Value) error {
 	return e.EncodeBytes(v.Bytes())
 }
 
+//------------------------------------------------------------------------------
+
+func encodeByteArrayValue(e *Encoder, v reflect.Value) error {
+	var bs []byte
+	if v.CanAddr() {
+		bs = v.Slice(0, v.Len()).Bytes()
+	} else {
+		bs = make([]byte, v.Len())
+		for i := range bs {
+			bs[i] = uint8(v.Index(i).Uint())
+		}
+	}
+	return e.EncodeBytes(bs)
```

Comment:
I see 2 possibilities:

- use reflect.Copy

```
	src := [3]byte{1, 2, 3}
	dst := make([]byte, len(src))
	n := reflect.Copy(reflect.ValueOf(dst), reflect.ValueOf(src))
	fmt.Println(n, dst)
```

I expect it to be faster, but would be nice to have a benchmark.

- don't create intermediate buffer, e.g.

```
e.EncodeBytesLen(v.Len())
for i := 0; i < v.Len(); i++ {
    e.w.WriteByte(v.Index(i).Uint())
}
```

Do you want to try it yourself? Let me know you want me to pick up this PR.